backend/src/main/java/com/bakdata/conquery/models/config/auth/AuthenticationConfig.java

```diff
@@ -1,20 +1,54 @@
 package com.bakdata.conquery.models.config.auth;
 
-import com.bakdata.conquery.commands.ManagerNode;
-import com.bakdata.conquery.io.cps.CPSBase;
-import com.bakdata.conquery.models.auth.ConqueryAuthenticationRealm;
+import com.bakdata.conquery.models.auth.web.AuthCookieFilter;
 import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
-
-@JsonTypeInfo(use = JsonTypeInfo.Id.CUSTOM, property = "type")
-@CPSBase
-public interface AuthenticationConfig {
-	
-	/**
-	 * Gets the realm specified in the configuration.
-	 * @return The realm.
-	 * @param managerNode
-	 */
+import io.dropwizard.util.Duration;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.Cookie;
+import javax.ws.rs.core.NewCookie;
+
+@NoArgsConstructor
+@Getter
+public class AuthenticationConfig {
+
+	// Define a maximum age since most browsers use session restoring making session cookies virtual permanent (see https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies)
+	@NotNull
+	private Duration adminEndCookieDuration = Duration.hours(1);
+
+	@JsonIgnore
+	private AuthCookieFilter authCookieFilter = null;
+
+
+
+
+	public Cookie createAuthCookie(ContainerRequestContext request, String token) {
+		return new NewCookie(
+				AuthCookieFilter.ACCESS_TOKEN,
+				token,
+				"/",
+				null,
+				0,
+				null,
+				Long.valueOf(adminEndCookieDuration.toSeconds()).intValue(),
+				null,
+				request.getSecurityContext().isSecure(),
+				true
+		);
+	}
+
 	@JsonIgnore
-	ConqueryAuthenticationRealm createRealm(ManagerNode managerNode);
+	public AuthCookieFilter getAuthCookieFilter() {
+		if (authCookieFilter == null) {
+			synchronized (this) {
```

Comment:
Ich bin mir nicht 100% sicher, aber könnte man so ein pattern sich nicht sparen in dem man eine AtomicReference mit java.util.concurrent.atomic.AtomicReference#updateAndGet benutzt?

Reply:
:D ich war dabei ein AtomicReference<T>.getOrInit(Supplier<T>) zu schreiben. Aber ich hab dann davon abgesehen mich in Atomic Reference einzulesen.
Bei updateAndGet musst du auch immer prüfen ob etwas drinne war